primal-sieve/src/streaming/mod.rs

```diff
@@ -352,4 +352,27 @@ mod tests {
             }
         }
     }
+
+    // These are designed to specifically test the medium sized and
+    // large prime sieving.
```

Comment:
These are dealing with `usize`, so your larger values should either be scaled back a little, or gate on `target_pointer_width = "64"`.

Reply:
IMO `deny(warnings)` is too annoying, as new rustc may add new warnings, but it's your call.

Reply:
This is a relatively small code base and deny(warnings) won't affect downstream users, so I'm happy to try it for a while, and revisit if it does get too annoying.